docs/setup/configuration.mdx

```diff
@@ -236,3 +240,24 @@ Saleor can automatically convert amounts between currencies. To use this feature
 ### `OPENEXCHANGERATES_API_KEY`
 
 Your store’s Open Exchange Rates "App ID".
+
+## Updating domain configuration
+
+When Saleor is exposed at a domain (e.g. `api.example.com`), it must be configured to generate URLs properly. Follow these steps to update the domain configuration:
+
+1. Use the GraphQL `shopDomainUpdate` mutation shown below to update Saleor's shop domain configuration with the new one.
+   ```graphql
+   mutation UpdateDomain {
+     shopDomainUpdate(input: { domain: "api.example.com" }) {
+       shop {
+         domain {
+           host
+         }
+       }
+     }
+   }
+   ```
+   Note that Saleor instances must be restarted for the new configuration to take effect, as shop data is cached in memory.
```

Comment:
@patrys @przlada is this note the only thing that blocks this PR from being merged ? Can we exclude this line and merge ?

Reply:
The note is accurate, meaning we're documenting poor architecture, but that's still better than not having documentation. A proper fix will come in https://github.com/saleor/saleor/issues/12748